cmd/juju/action/run.go

```diff
@@ -133,53 +133,54 @@ func (c *runCommand) SetFlags(f *gnuflag.FlagSet) {
 func (c *runCommand) Info() *cmd.Info {
 	return &cmd.Info{
 		Name:    "run-action",
-		Args:    "<unit> <action name> [key.key.key...=value]",
+		Args:    "<unit> [<unit> ...] <action name> [key.key.key...=value]",
 		Purpose: "Queue an action for execution.",
 		Doc:     runDoc,
 	}
 }
 
-// Init gets the unit tag, and checks for other correct args.
+// Init gets the unit tag(s), action name and action arguments.
 func (c *runCommand) Init(args []string) error {
-	switch len(args) {
-	case 0:
+	var unitNames []string
+	for idx, arg := range args {
+		if names.IsValidUnit(arg) {
```

Comment:
Once I was able to call an action against multiple units, I was going to add 'juju run-action --leader foo my-action' and 'juju run-action --all foo my-action'. The new flags mean the second argument is the application name, which is then used to populate the list of units to run the action on.

I could make it look more like the existing 'juju run' cli. The alternative is to make the 'juju run' cli work like what is proposed here ;-) I could argue that '--units' is unnecessary, as is requiring the list of units to be comma separated. Although comma separated would make sense if we wanted to run actions on all units or lead units of multiple applications, as then we can avoid a namespace clash with the action name.

juju run-action foo/0 act-name param=value vs juju run-action --units foo/0 act-name param=value
juju ru-action foo/0 foo/1 act-name

Collapsing run and run-action into one command is too invasive for me to tackle properly. The run-action command would need to remain for backwards compatibility and the asynchronous cli (assuming we take the opportunity to make synchronous the default for the cli)